### PR TITLE
Suggestions on improving code and the solution for the issue#10

### PR DESCRIPTION
**Pull request by shiv-chan**
@Dannyboy222 I updated your code for the header component here.

what I changed from your code are...
- Installed [`hamburger-react`](https://hamburger-react.netlify.app/) to create the hamburger menu button instead of using a svg image.
- Wrapped the header bar with `<header>` tag and placed the menu overlay aside.
- Use a custom class for header height (as we discussed and set up in `tailwind.config.js`)
- Removed unused or duplicated custom class in `tailwind.config.js`
- Set a function that enable the user to close the menu overlay by clicking outside of it.

🚨  haven't updated the responsive version yet!

### CODE REVIEW

File: client/src/common/components/Header.tsx [hunk 1]
```diff
@@ -1,161 +1,86 @@
...
+import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
-// import { FaTimes } from 'react-icons/fa';
-// import { FaBars } from 'react-icons/fa';
+import { Squash as Hamburger } from 'hamburger-react';
 
 const Header = () => {
-	const [isClosed, setClosed] = useState(true);
+	const [isOpen, setOpen] = useState(false);
+	const [isLogin, setIsLogin] = useState(false); // TODO: replace with a global state later
```
Comment (shiv-chan) [inline client/src/common/components/Header.tsx:7]: Please change `true`, `false` manually to see different versions.

**Review by Dannyboy222 — APPROVED** [review]
looks amazing to me!

File: client/src/common/components/Header.tsx [hunk 2]
```diff
@@ -1,161 +1,86 @@
...
-			{/* )} */}
+	const closeClickingOutsideMenu = (e: any) => {
+		if (e.target.closest('.menu') === null) {
+			setOpen(false);
+		}
+	};
+
+	useEffect(() => {
+		document.addEventListener('click', closeClickingOutsideMenu);
+	}, []);
```
Comment (miyabitanimchi) [inline client/src/common/components/Header.tsx:53]: I don't think we need to use document.addEventlistener() on React. Instead we can just use onClick on elements.
You can check [the official document](https://reactjs.org/docs/handling-events.html#:~:text=When%20using%20React%2C%20you%20generally,the%20element%20is%20initially%20rendered.&text=You%20have%20to%20be%20careful,are%20not%20bound%20by%20default.)
And with this code `closeClickingOutsideMenu()` always gets executed every time a user clicks on a page even though the menu is closed, and maybe this is a little problem.
To solve this, we can create an element as kind of overlay just under the dropdown menu. (image below).
But what do you think?
Image from iOS (3)

**Review by miyabitanimchi — COMMENTED** [review]
Thank you! it worked well and I loved the animation too.
But there is one thing I noticed on the code, so please check the comment I left.

File: client/src/common/components/Header.tsx (hunk 2 above)
Comment (shiv-chan) [reply client/src/common/components/Header.tsx:53]: > When using React, you generally don’t need to call addEventListener to add listeners to a DOM element after it is created. Instead, just provide a listener when the element is initially rendered.

Probably you pointed out this one on the doc.  I intended to provide a listener when the first render occurs using `useEffect` with no independencies as this part says.

> And with this code `closeClickingOutsideMenu()` always gets executed every time a user clicks on a page even when the menu is closed

I agree with you. Could you add the overlay part for us, please?
Maybe the overlay has to move together with the menu.
We can talk about this later.

Thanks for reviewing thoroughly 🙏

**Pull request merged**